### PR TITLE
PT-1895: Localize countries in UI

### PR DESCRIPTION
**Pull request by eMazeika**
https://virtocommerce.atlassian.net/browse/PT-1895

### CODE REVIEW

**Review by akak1977 — APPROVED** [review]

File: src/VirtoCommerce.Platform.Web/wwwroot/js/i18n/resources/countries.js [hunk 1]
```diff
@@ -1,273 +1,281 @@
...
+            },
+            normalize: function (id) {
+                var result = undefined;
+                if (!!id) {
+                    result = id.toUpperCase();
+                }
+                return result;
+            },
+            query: function () {
+                var results = [
```
Comment (Vectorfield4) [inline src/VirtoCommerce.Platform.Web/wwwroot/js/i18n/resources/countries.js:20]: Why this list is not deleted from the platform?

**Review by Vectorfield4 — COMMENTED** [review]
Why this file is still in the platform `src/VirtoCommerce.Platform.Web/wwwroot/js/i18n/resources/countries.js`? For backward compatibility?

File: src/VirtoCommerce.Platform.Web/wwwroot/js/i18n/resources/countries.js (hunk 1 above)
Comment (eMazeika) [reply src/VirtoCommerce.Platform.Web/wwwroot/js/i18n/resources/countries.js:20]: It's still the primary source of countries. Any requirement to refactor?

Comment (Vectorfield4) [reply src/VirtoCommerce.Platform.Web/wwwroot/js/i18n/resources/countries.js:20]: image
With this requirement, we don't this list in the JS file.

Comment (tatarincev) [reply src/VirtoCommerce.Platform.Web/wwwroot/js/i18n/resources/countries.js:20]: I agree with @Vectorfield4, why do we still need the hard-coded country list in this file? Why don't you move the list of countries for the English language to the default resource file, and read it as an array here instead?

Comment (tatarincev) [reply src/VirtoCommerce.Platform.Web/wwwroot/js/i18n/resources/countries.js:20]: I'm pretty sure $translate has the ability to read resource arrays for a specific path. If not, we can always read localization resources as JSON.

Comment (eMazeika) [reply src/VirtoCommerce.Platform.Web/wwwroot/js/i18n/resources/countries.js]: API added

File: src/VirtoCommerce.Platform.Web/Controllers/Api/CommonController.cs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        /// </summary>
+        [HttpGet]
+        [Route("countries")]
+        [AllowAnonymous]
+        public ActionResult<List<Country>> GetCountries()
+        {
+            var cacheKey = CacheKey.With(GetType(), nameof(GetCountries));
+            var results = _memoryCache.GetOrCreateExclusive(cacheKey, (cacheEntry) =>
+            {
+                var filePath = Path.GetFullPath("app_data/countries.json");
```
Comment (Vectorfield4) [inline src/VirtoCommerce.Platform.Web/Controllers/Api/CommonController.cs:36]: Maybe we need a possibility to load countries from URL from app settings.

**Review by Vectorfield4 — APPROVED** [review]

File: src/VirtoCommerce.Platform.Web/Controllers/Api/CommonController.cs (hunk 2 above)
Comment (eMazeika) [reply src/VirtoCommerce.Platform.Web/Controllers/Api/CommonController.cs:36]: it should be ok for now

**Pull request merged**